### PR TITLE
dust attenuation curves addition

### PR DESCRIPTION
**Pull request by LauraSommovigo**
I added another parametric formula for the attenuation curve, from Li+08. This formula has been applied to JWST data in Markov+23, deriving the best-fitting parameters for a few z/sim 7 galaxies (Markov+23) and a complete JWST-selected sample of z=2-12 galaxies (Markov in prep.). NB: there might be a problem with the LMC and SMC attenuation curve in the current "GrainsWD01" model. For instance, the 2175 angstrom bump seems to be more pronounced in the LMC than in the MW curve. Maybe check it?

## Issue Type

- Bug
- Document
- Enhancement

## Checklist
- [x] I have read the [CONTRIBUTING.md]() -->
- [x] I have added docstrings to all methods
- [x] I have added sufficient comments to all lines
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no pep8 errors
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: attenuation.py [hunk 1]
```diff
@@ -0,0 +1,513 @@
...
+from unyt import Angstrom, unyt_quantity, unyt_array
+
+from dust_extinction.grain_models import WD01
+
+from synthesizer import exceptions
+
+
+this_dir, this_filename = os.path.split(__file__)
+
+__all__ = ["PowerLaw", "MW_N18", "Calzetti2000", "GrainsWD01", "LMC_M23"]
```
Comment (aswinpvijayan) [inline attenuation.py:13]: LMC_M23 is not the class name currently implemented

**Comment by aswinpvijayan** [comment]
If you could please add the GrainsWD01 LMC model as a separate issue that would be great.

**Comment by aswinpvijayan** [comment]
Its currently failing the check the build, since the class name is different. Could you change it to the actual name "Markov23". 
Also it would be benificial to have option for the user to automatically have the different 'c' values by having an option to call the MW, SMC or LMC curve.

File: attenuation.py [hunk 2]
```diff
@@ -0,0 +1,513 @@
...
+
+def N09_tau(lam, slope, cent_lam, ampl, gamma):
```
Comment (WillJRoper) [inline attenuation.py:15]: This is hugely pedantic so I apologise, but please use snake_case for functions and variables and reserve CamelCase for class names. (This does need fixing in a bunch of places elsewhere in the code but I think I'm getting to the end of the list)

File: attenuation.py [hunk 3]
```diff
@@ -0,0 +1,513 @@
...
+        """
+        if isinstance(lam, (unyt_quantity, unyt_array)):
+            _lam = lam.to('Angstrom').v
+        else:
+            _lam = lam
+        return self.emodel.extinguish(x=(_lam*Angstrom).to_astropy(), Av=1.086 * tau_v)
+
+
+
+def Li_08(lam, c1, c2, c3, c4):
```
Comment (WillJRoper) [inline attenuation.py:425]: Same here

File: attenuation.py [hunk 4]
```diff
@@ -0,0 +1,513 @@
...
+        # Wavelength in microns
+        if isinstance(lam, (unyt_quantity, unyt_array)):
+            lam_micron = lam.to('um').v
+        else:
+            lam_micron = lam/1e4
+            
+        
+        
+        #Attenuation curve normalized to Av
+        A_lam_v = c1 / ((lam_micron/0.08)**c2 + (lam_micron/0.08)**-c2 + c3)  +   (233. * (1 - c1/(6.88**c2 + 0.145**c2 +c3) - c4/4.6)) / ((lam_micron/0.046)**2. + (lam_micron/0.046)**-2. + 90.)  +  c4/ ((lam_micron/0.2175)**2. + (lam_micron/0.2175)**-2. - 1.95)
```
Comment (WillJRoper) [inline attenuation.py:459]: And here

Comment (WillJRoper) [reply attenuation.py:459]: Could this line be broken up for readability? (And keeping to 79 characters per line).

File: attenuation.py [hunk 5]
```diff
@@ -0,0 +1,513 @@
...
+
+    Attributes:
+            c1, c2, c3, c4 (float)
+                Dimensionless parameters describing the FUV rise, optical/NIR regime and 2175 Angstrom bump of the attenuation curve.
+
+    """
+
+    def __init__(self, c1=1., c2=1., c3=1., c4=0.):
+        """
+        Initialise the dust curve.
```
Comment (WillJRoper) [inline attenuation.py:478]: Missing the argument description from the doc string here. Could these parameters be better named to encode which is relevant to what part of the spectrum? I have zero knowledge of the model so I don't know 😅

**Review by WillJRoper — CHANGES_REQUESTED** [review]
Looks good! A few issues and comments to resolve, the main one being the test failure due to the issue @aswinpvijayan raised. If you want more details you can see the error by looking at the failed action (click the failed build on the PR).

Sorry for the pedantic style comments from me. The file lists every line as added (not sure why) so its possible I've highlighted stuff that wasn't written by you. We're trying to get everything in order and I'm a stickler for code style 😅.

File: attenuation.py (hunk 1 above)
Comment (christopherlovell) [reply attenuation.py:13]: Fixed

**Review by christopherlovell — APPROVED** [review]

**Review by christopherlovell — APPROVED** [review]

**Pull request merged**